src/main/java/com/chrosciu/bootcamp/tasks/github/GithubClient.java

```diff
@@ -4,28 +4,41 @@
 import com.chrosciu.bootcamp.tasks.github.dto.Repository;
 import lombok.RequiredArgsConstructor;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
 
 @RequiredArgsConstructor
 public class GithubClient {
     private final GithubApi githubApi;
 
     public Flux<Repository> getUserRepositories(String username) {
-        //TODO: Implement
-        return null;
+        Mono<List<Repository>> mono = githubApi.getUserRepositories(username);
+        Flux<Repository> flux = mono.flatMapMany(Flux::fromIterable);
+        return flux;
     }
 
     public Flux<Branch> getUserRepositoryBranches(String username, String repo) {
-        //TODO: Implement
-        return null;
+
+        Mono<List<Branch>> mono = githubApi.getUserRepositoryBranches(username, repo);
+        Flux<Branch> flux = mono.flatMapMany(Flux::fromIterable);
+        return flux;
     }
 
     public Flux<Repository> getUsersRepositories(Flux<String> usernames) {
-        //TODO: Implement
-        return null;
+
+    /*  Flux<Repository> flux = usernames.flatMap(x -> getUserRepositories(x));
+        return flux;*/
+        return usernames.flatMap(user -> getUserRepositories(user));
     }
 
     public Flux<String> getAllUserBranchesNames(String username) {
-        //TODO: Implement
-        return null;
+
+        Flux<Repository> flux = getUserRepositories(username);
```

Comment:
Całkiem przyzwoicie, ja bym to pewnie zapisał w postaci od razu całego pociągu (flatMap().map()..) ale nie ma w tym nic złego - tworzenie tych zmiennych jest praktycznie bezkosztowe (strumienie są bowiem leniwe)